util/httputil/http_util.go

```diff
@@ -144,38 +161,99 @@ func CreateTransport(conf *config.Config, auth *config.Auth, transportConfig *ht
 	outerRoundTripper := newCustomHeadersRoundTripper(customHeaders, transportConfig)
 
 	if auth != nil {
-		tlscfg, err := GetTLSConfig(conf, auth)
-		if err != nil {
-			return nil, err
-		}
-		if tlscfg != nil {
-			transportConfig.TLSClientConfig = tlscfg
+		transportConfig.DialTLSContext = func(ctx context.Context, network, addr string) (net.Conn, error) {
+			host, _, err := net.SplitHostPort(addr)
+			if err != nil {
+				host = addr
+			}
+			cfg, err := buildTLSConfig(conf, auth, host)
+			if err != nil {
+				return nil, err
+			}
+			dialer := tls.Dialer{Config: cfg}
+			return dialer.DialContext(ctx, network, addr)
 		}
-		outerRoundTripper = newAuthRoundTripper(auth, outerRoundTripper)
+		outerRoundTripper = newAuthRoundTripper(conf, auth, outerRoundTripper)
 	}
 
 	return outerRoundTripper, nil
 }
 
 func GetTLSConfig(conf *config.Config, auth *config.Auth) (*tls.Config, error) {
-	if auth.InsecureSkipVerify || auth.CAFile != "" {
-		certPool := conf.CertPool()
-		if auth.CAFile != "" {
-			cert, err := os.ReadFile(auth.CAFile)
+	if auth == nil {
+		return nil, nil
+	}
+	if auth.CertFile == "" && auth.KeyFile == "" && !auth.InsecureSkipVerify {
+		return nil, nil
+	}
+	return buildTLSConfig(conf, auth, "")
+}
+
+// GetTLSConfigForServer returns a tls.Config that is pre-configured for the provided server name.
+// This should be preferred when the target hostname/IP is known ahead of time (e.g. for gRPC clients)
+// so that certificate verification can enforce the correct SAN.
+func GetTLSConfigForServer(conf *config.Config, auth *config.Auth, serverName string) (*tls.Config, error) {
+	if auth == nil {
+		return nil, nil
+	}
+	return buildTLSConfig(conf, auth, serverName)
+}
+
+func buildTLSConfig(conf *config.Config, auth *config.Auth, serverName string) (*tls.Config, error) {
+	cfg := &tls.Config{
+		MinVersion: tls.VersionTLS12,
```

Comment:
This might be a problem for FIPS enabled clusters?

Reply:
This feels like the kind of thing that will likely break someone if it's not configurable.

Reply:
We've never had this configurable before, and I'm fairly confident no one is going to want to allow TLS 1.1 or lower. (e.g. major browsers dropped TLS 1.0/1.1 support in 2020).

They may want it configurable HIGHER - but we never were able to fix on a specific minimum TLS version before (that I know of), so this just hardens the server. We could introduce a TLS version minimum configuration setting later as a separate issue. There are cases I can see where they may want a minimum of 1.3.